src/components/LinkedAccounts/ConfirmationModal/LinkedAccountToConfirm.tsx

```diff
@@ -1,41 +1,114 @@
-import React from 'react'
+import React, { forwardRef, useImperativeHandle, useState } from 'react'
+import { isEqual, startOfDay } from 'date-fns'
+import classNames from 'classnames'
 import type { LinkedAccount } from '../../../types/linked_accounts'
 import { Heading } from '../../ui/Typography/Heading'
 import { P } from '../../ui/Typography/Text'
-import { VStack } from '../../ui/Stack/Stack'
+import { VStack, Stack } from '../../ui/Stack/Stack'
 import { Checkbox } from  '../../ui/Checkbox/Checkbox'
+import { InputGroup } from '../../Input'
+import { AmountInput } from '../../Input/AmountInput'
+import { DatePicker } from '../../DatePicker'
+import InstitutionIcon from '../../../icons/InstitutionIcon'
 
-type LinkedAccountConfirmationProps = {
+export type LinkAccountForm = {
   account: LinkedAccount
   isConfirmed: boolean
-  onChangeConfirmed: (isConfirmed: boolean) => void
+  openingDate?: Date
+  openingBalance?: number
+}
+
+export type LinkAccountToConfirmRef = {
+  getData: () => LinkAccountForm
+}
+
+type LinkedAccountConfirmationProps = {
+  account: LinkedAccount
+  defaultValue: LinkAccountForm
+  compact?: boolean
+  disableConfirmExclude?: boolean
 }
 
 const CLASS_NAME = 'Layer__LinkedAccountToConfirm'
 
-export function LinkedAccountToConfirm({
+const LinkedAccountToConfirm = forwardRef(({
```

Comment:
I had to use `forwardRef` and `useImperativeHandle` to update parent's form state when form is being submitted instead of updating on every change. Otherwise, the parent state was reconstructed, components re-rendered, and input was losing focus.